include/zephyr/storage/flash_map.h

```diff
@@ -106,6 +106,8 @@ int flash_area_check_int_sha256(const struct flash_area *fa,
 				const struct flash_area_check *fac);
 #endif
 
+#if defined(CONFIG_FLASH_MAP)
```

Comment:
Oh no. If entire API header has to be conditional, maybe it should not be included in the first place.

Reply:
fixed